File: server/router/noc/noc.go
package noc

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type NocRouter struct{}

func (s *NocRouter) InitNocRouter(Router *gin.RouterGroup) {
	nocRouter := Router.Group("noc").Use(middleware.OperationRecord())
	nocApi := v1.ApiGroupApp.NocApiGroup.NocApi
	{
		nocRouter.POST("createNoc", nocApi.CreateNoc)
		nocRouter.DELETE("deleteNocById", nocApi.DeleteNocById)
		nocRouter.PUT("updateNocById", nocApi.UpdateNocById)
	}
}

func (s *NocRouter) InitPubNocRouter(Router *gin.RouterGroup) {
	nocRouterWithoutRecord := Router.Group("noc")
	nocApi := v1.ApiGroupApp.NocApiGroup.NocApi
	{
		nocRouterWithoutRecord.GET("getNocList", nocApi.GetNocList)
		nocRouterWithoutRecord.GET("getNocDetailById", nocApi.GetNocDetailById)
		nocRouterWithoutRecord.GET("getNocById", nocApi.GetNocById)
	}
}

File: web/src/api/noc.js
import service from '@/utils/request'

export const createNoc = (data) => {
  return service({
    url: '/noc/createNoc',
    method: 'post',
    data
  })
}

export const deleteNocById = (data) => {
  return service({
    url: '/noc/deleteNocById',
    method: 'delete',
    data
  })
}

export const updateNocById = (data) => {
  return service({
    url: '/noc/updateNocById',
    method: 'put',
    data
  })
}

export const getNocDetailById = (params) => {
  return service({
    url: '/noc/getNocDetailById',
    method: 'get',
    params
  })
}

export const getNocById = (params) => {
  return service({
    url: '/noc/getNocById',
    method: 'get',
    params
  })
}

export const getNocList = (params) => {
  return service({
    url: '/noc/getNocList',
    method: 'get',
    params
  })
}
